lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1638,6 +1638,7 @@ sub _failure_reason {
 
     while (my $m = $modules->next) {
         if ($m->result eq FAILED || $m->result eq SOFTFAILED) {
+            last unless $m->results;
```

Comment:
These variable names are just confusing. What about:

```
last unless my $results = $m->results;
last unless my $details = $results->{details};
```
And then use `$details` instead of `$results` in the subsequent code.

Reply:
right, because that just changed due to your recent merge of another PR :) Fixed

Reply:
Like you've fixed it now it makes no sense. I mean the method is still called `results`. Why not writing it in the way I suggested?

Reply:
And I assume we still need to fix the case when `results` returns `undef`.

Reply:
Right, that was stupid. Sorry for wasting your time :D
Tested now locally with `prove -l  t/api/02-iso-download.t t/23-amqp.t t/40-script_openqa-clone-custom-git-refspec.t t/10-jobs.t t/20-stale-job-detection.t`. Updated.